src/main/java/org/spongepowered/common/service/sql/SqlServiceImpl.java

```diff
@@ -134,7 +134,11 @@ public HikariDataSource load(@Nonnull ConnectionInfo key) throws Exception {
                     if (driverSpecificProperties != null) {
                         config.setDataSourceProperties(driverSpecificProperties);
                     }
-                    config.setJdbcUrl(key.getAuthlessUrl());
+                    if (key.getDriverClassName().equals("com.mysql.cj.jdbc.Driver")) {
+                        config.setJdbcUrl(key.getAuthlessUrl() + "?disableMariaDbDriver");
```

Comment:
Info for anyone reviewing / testing: https://mariadb.com/kb/en/library/about-mariadb-connector-j/#having-mariadb-and-mysql-drivers-in-the-same-classpath